src/rotator_library/ip_throttle_detector.py

```
@@ -0,0 +1,387 @@
+        """
+        records = self._records[provider]
+
+        if not records:
+            return ThrottleAssessment(
+                scope=ThrottleScope.CREDENTIAL,
+                confidence=1.0,
+                suggested_cooldown=self.credential_cooldown,
+            )
+
+        # Get unique credentials
+        unique_credentials = list(set(r.credential for r in records))
```

Ruff (0.15.4)

[convention] Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)